models/enum_type.go

```diff
@@ -10,6 +11,22 @@ type EnumType struct {
 	Values   []interface{}
 }
 
+// Ensure EnumType implements the Equaler interface
+var _ Equaler = EnumType{}
+var _ Equaler = (*EnumType)(nil)
+
+// Equal returns true if two EnumType objects are equal; otherwise false is returned.
+func (self EnumType) Equal(u Equaler) bool {
+	other := u.(EnumType)
```

Comment:
@tsmaeder just found out that it doesn't crash immediately. We can have a type assertion by using the second value:

```go
other, ok := u.(EnumType)
if !ok {
    return false
}
```